source/video_core/src/core/hw/gpu.cpp

```diff
@@ -288,7 +291,7 @@ inline void Write(Pica::Context& context, u32 addr, const T data) {
                             switch (ToGenericFormat(config.GetInputFormat())) {
                             case GenericImageFormat::RGBA8:
                             {
-                                PAddr pixel_addr = x * 4 * pixel_skip + y * pixel_skip_y * config.display_transfer.input_width * 4;
+                                PAddr pixel_addr = x * 4 * pixel_skip + flipped_y * pixel_skip_y * config.display_transfer.input_width * 4;
```

Comment:
There are some open questions here regarding how specifically the flipping should operate: Does the input image get flipped and then copied (as done currently), or does the input image get copied onto a mirrored output area?

This only becomes relevant if `output_height != input_height`. We can avoid clarifying this for now by strengthening the validation checks above. if `flip_y` is set, could we strengthen the `if (config.display_transfer.output_height > config.display_transfer.input_height) {` check above to use `==`? (Assuming the games you tested still work then.)

Reply:
If you mean adding the following 'if' below the one you mentioned, then yes it still works with Yokai Watch and other Level5 games.
```
if (config.flip_data && (config.display_transfer.output_height != config.display_transfer.input_height)) {
    throw std::runtime_error(fmt::format(   "Input height {:#x} doesn't match output height {:#x} exactly",
        config.display_transfer.input_height.Value(),
        config.display_transfer.output_height.Value()));
}
```

Reply:
`"Flipping Y; Input height {:#x} doesn't match output height {:#x} exactly"` is a bit more descriptive, so will use that message instead.